huaweicloud/resource_huaweicloud_cce_cluster_v3.go

```diff
@@ -238,6 +258,25 @@ func resourceClusterExtendParamV3(d *schema.ResourceData, config *Config) map[st
 	return m
 }
 
+func resourceClusterMastersV3(d *schema.ResourceData) ([]clusters.MasterSpec, error) {
+	flavorId := d.Get("flavor_id").(string)
+	mastersRaw := d.Get("masters").([]interface{})
+	if strings.Split(flavorId, ".")[1] == "s1" && len(mastersRaw) != 1 {
+		return []clusters.MasterSpec{}, fmt.Errorf("Error creating HuaweiCloud Cluster: single-master cluster need 1 az for master node, but got %d", len(mastersRaw))
+	}
+	if strings.Split(flavorId, ".")[1] == "s2" && len(mastersRaw) != 3 {
+		return []clusters.MasterSpec{}, fmt.Errorf("Error creating HuaweiCloud Cluster: high-availability cluster need 3 az for master nodes, but got %d", len(mastersRaw))
```

Comment:
it's better to return *nil, error* when the validation failed.

Reply:
fixed